### PR TITLE
PCHR-4327: Create option groups for organisation contact in job contract

### PR DESCRIPTION
**Pull request by ajesamson**
## Overview
For setting up insurance or pension providers, organisation contact records were created for such. This PR creates an option group for the purpose of managing providers list.

## Before
There were no option group for managing organisation contacts used for insurance or pension.

## After
Three option groups were created, each for managing organisation contacts belonging to Health Insurance Provider, Life Insurance Provider, and Pension Provider

## Technical Details
An XML file was created for setting up insurance and pension option groups. This was then picked up by an upgrader for creating the required option groups.
```
if ($result['count'] === 0) {
  $file = 'xml/organisation_contact_install.xml';
  $this->executeCustomDataFile($file);
}
```

### CODE REVIEW

File: hrjobcontract/xml/organisation_contact_install.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="iso-8859-1" ?>
+
+<CustomData>
+
```
Comment (tunbola) [inline hrjobcontract/xml/organisation_contact_install.xml]: We probably don't need this extra space. Maybe an extra space between the option group items would be good.

File: hrjobcontract/xml/organisation_contact_install.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="iso-8859-1" ?>
```
Comment (tunbola) [inline hrjobcontract/xml/organisation_contact_install.xml]: Now might be a good time to move the xml files for option groups into their own folder. i.e create an option_group folder within the xml folder and move the xml option groups related files there. I took a quick look and it seems the files are referenced in one or two places, so shouldn't be a big change.

File: hrjobcontract/CRM/Hrjobcontract/Upgrader.php [hunk 3]
```diff
@@ -1053,6 +1053,29 @@ public function upgrade_1043() {
...
+   * Health Insurance Provider, Life Insurance Provider, or Pension Provider
+   *
+   * @return bool
+   */
+  public function upgrade_1044() {
+    $result = civicrm_api3('OptionGroup', 'get', [
+      'name' => ['IN' => [
+        'hrjc_health_insurance_provider',
+        'hrjc_life_insurance_provider',
+        'hrjc_pension_providers'
```
Comment (tunbola) [inline hrjobcontract/CRM/Hrjobcontract/Upgrader.php]: I noticed that the first two option group are singular while the third is plural. We should stick to one. Singular should work for all.

File: hrjobcontract/CRM/Hrjobcontract/Upgrader.php [hunk 4]
```diff
@@ -1053,6 +1053,29 @@ public function upgrade_1043() {
     return TRUE;
   }
 
+  /**
+   * Creates option group for organisation contact belonging to sub types
```
Comment (tunbola) [inline hrjobcontract/CRM/Hrjobcontract/Upgrader.php]:
```suggestion
   * Creates option groups for organisation contacts belonging to sub types
```

File: hrjobcontract/xml/organisation_contact_install.xml (hunk 2 above)
Comment (ajesamson) [reply hrjobcontract/xml/organisation_contact_install.xml]: Alright. I actually did that first but reverted to maintain conformity with previous implementations. 👍

File: hrjobcontract/CRM/Hrjobcontract/Upgrader.php [hunk 5]
```diff
@@ -1064,12 +1065,12 @@ public function upgrade_1044() {
...
         'hrjc_health_insurance_provider',
         'hrjc_life_insurance_provider',
-        'hrjc_pension_providers'
+        'hrjc_pension_provider'
       ]],
     ]);
 
     if ($result['count'] === 0) {
-      $file = 'xml/organisation_contact_install.xml';
+      $file = 'xml/option_groups/organisation_contact_install.xml';
```
Comment (tunbola) [inline hrjobcontract/CRM/Hrjobcontract/Upgrader.php]: We should rename this to organization_provider too.

File: hrjobcontract/CRM/Hrjobcontract/Upgrader.php [hunk 6]
```diff
@@ -1054,7 +1055,7 @@ public function upgrade_1043() {
   }
 
   /**
-   * Creates option group for organisation contact belonging to sub types
+   * Creates option groups for organisation contacts belonging to sub types
```
Comment (tunbola) [inline hrjobcontract/CRM/Hrjobcontract/Upgrader.php]: s/organisation contacts/organisation providers

**Pull request merged**